r-package/grf/R/quantile_forest.R

```diff
@@ -177,18 +178,20 @@ quantile_forest <- function(X, Y,
 #' @export
 predict.quantile_forest <- function(object,
                                     newdata = NULL,
-                                    quantiles = c(0.1, 0.5, 0.9),
+                                    quantiles = NULL,
                                     num.threads = NULL, ...) {
-  if (!is.numeric(quantiles) | length(quantiles) < 1) {
-    stop("Error: Must provide numeric quantiles")
-  } else if (min(quantiles) <= 0 | max(quantiles) >= 1) {
-    stop("Error: Quantiles must be in (0, 1)")
-  }
-
-  # If possible, use pre-computed predictions.
-  quantiles.orig <- object[["quantiles.orig"]]
-  if (is.null(newdata) & identical(quantiles, quantiles.orig) & !is.null(object$predictions)) {
-    return(object$predictions)
+  if (is.null(quantiles)) {
+    quantiles <- object[["quantiles.orig"]]
+    # If possible, use pre-computed predictions.
```

Comment:
Small comment, for me it'd be clearer to keep this as before, outside of this conditional:

```
if (is.null(newdata) & identical(quantiles, quantiles.orig) & !is.null(object$predictions)) {	      
    return(object$predictions)
}
```

That way we have the invariant 'if the training and prediction quantiles are the same, then we use pre-computed OOB values'.